Launcher/src/update.c

```diff
@@ -0,0 +1,139 @@
+#include <archive.h>
+#include <archive_entry.h>
+#include <assert.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <sys/stat.h>
+
+#include "launcher.h"
+
+int endsWith(const char *str, const char *end) {
+    size_t lenStr = strlen(str);
+    size_t lenEnd = strlen(end);
+    if (lenEnd >  lenStr) {
+        return 0;
+	}
+    return strncmp(str + lenStr - lenEnd, end, lenEnd) == 0;
+}
+
+void removeFiles(char *path) {
+	DIR *dir;
+	struct dirent *ent;
+	struct stat s;
+	if ((dir = opendir(path)) != NULL) {
+		while ((ent = readdir(dir)) != NULL) {
+			char *current = malloc(strlen(path) + sizeof(ent->d_name) + 3);
+            sprintf(current, "%s%s%s", path, DELIM, ent->d_name);
+			stat(current, &s);
+
+			if (strcmp(ent->d_name, ".") != 0 &&
```

Comment:
If I understand correctly, this will purge all jar files, not only the default ones (ResourceAnalyzer.jar, DefaultMusic.jar,...).